src/app/dataset-view/dataset.component.ts

```diff
@@ -34,64 +34,14 @@ import { Dataset, DatasetKind } from "../api/kamu.graphql.interface";
 export class DatasetComponent implements OnInit, OnDestroy {
```

Comment:
The following fields and related behavior repeat between many components:
```
@ViewChild("sidenav", { static: true }) public sidenav?: MatSidenav;
public isMinimizeSearchAdditionalButtons = false;
```

Also a few more fields/behaviors repeat between main window components - `SearchComponent` and `DatasetComponent`:

```
public isMobileView = false;
public searchValue = "";
private w: Window;
```
Maybe new `subs` would also belong to these categories.

I think we should stop copy pasting these, and instead create a hierarchy of base abstract components. Perhaps 2 levels:
 - very general base (shared between all components)
 - main window base (extends the one above, generalizes common parts of `DatasetComponent` and `SearchComponent`)
 
The ultimate goal is not to have duplicate peripheral functionality between components. They should concentrate on their purpose instead and share utilities in base classes and/or utility components.

Reply:
I need to move sidenav in separate shared component. Because we use it in many places.